src/main/kotlin/no/nav/familie/ba/sak/behandling/vedtak/VedtakService.kt

```diff
@@ -87,21 +100,119 @@ class VedtakService(private val behandlingService: BehandlingService,
                                       fom = periode.fom,
                                       tom = periode.tom)
 
-        vedtak.leggTilUtbetalingBegrunnelse(begrunnelse)
+        vedtak.leggTilBegrunnelse(begrunnelse)
 
         lagreEllerOppdater(vedtak)
 
         return vedtak.utbetalingBegrunnelser.map {
-            it.tilRestUtbetalingBegrunnelse()
+            it.tilRestVedtakBegrunnelse()
+        }
+    }
+
+    @Transactional
+    fun leggTilBegrunnelse(restPostVedtakBegrunnelse: RestPostVedtakBegrunnelse,
+                           fagsakId: Long): List<UtbetalingBegrunnelse> {
+
+        val vedtakBegrunnelseType = restPostVedtakBegrunnelse.vedtakBegrunnelse.vedtakBegrunnelseType
+        val vedtakBegrunnelse = restPostVedtakBegrunnelse.vedtakBegrunnelse
+
+        val vedtak = hentVedtakForAktivBehandling(fagsakId) ?: throw Feil(message = "Finner ikke aktiv vedtak på behandling")
+
+        val personopplysningGrunnlag = persongrunnlagService.hentAktiv(vedtak.behandling.id)
+                                       ?: throw Feil("Finner ikke personopplysninggrunnlag ved fastsetting av begrunnelse")
+
+
+        val vilkårsvurdering = vilkårsvurderingService.hentAktivForBehandling(vedtak.behandling.id)
+                               ?: throw Feil("Finner ikke vilkårsvurdering ved fastsetting av begrunnelse")
+
+        val personerMedUtgjørendeVilkårForUtbetalingsperiode =
+                hentPersonerMedUtgjørendeVilkår(
+                        vilkårsvurdering = vilkårsvurdering,
+                        utbetalingsperiode = Periode(
+                                fom = restPostVedtakBegrunnelse.fom,
+                                tom = restPostVedtakBegrunnelse.tom
+                        ),
+                        oppdatertBegrunnelseType = vedtakBegrunnelseType,
+                        utgjørendeVilkår = vedtakBegrunnelse.finnVilkårFor())
+
+        val barnaMedVilkårSomPåvirkerUtbetaling = personerMedUtgjørendeVilkårForUtbetalingsperiode.filter {
+            it.first.type == PersonType.BARN
+        }.map {
+            it.first
+        }
+
+        val barnasFødselsdatoer = slåSammen(barnaMedVilkårSomPåvirkerUtbetaling.sortedBy { it.fødselsdato }
+                                                    .map { it.fødselsdato.tilKortString() })
+
+        val brevBegrunnelse = if (VedtakBegrunnelseSerivce.utenVilkår.contains(vedtakBegrunnelse)) {
+            if (vedtakBegrunnelse == VedtakBegrunnelse.INNVILGET_SATSENDRING
+                && SatsService.finnSatsendring(restPostVedtakBegrunnelse.fom).isEmpty()) {
+                throw FunksjonellFeil(melding = "Begrunnelsen stemmer ikke med satsendring.",
+                                      frontendFeilmelding = "Begrunnelsen stemmer ikke med satsendring. Vennligst velg en annen begrunnelse.")
+            }
```

Comment:
```suggestion
sjekkOmSatsEndring(vedtakBegrunnelse, restPostVedtakBegrunnelse)
```

leggTilBegrunnelse begynner å bli veldig stor. Hadde den blitt litt mer lesbar om vi hadde pakka inn noen av sjekkene i små funksjoner? :)

Reply:
mm, enig. Ikke helt scope for oppgaven(har egentlig bare forenklet endreBegrunnelse->leggTilBegrunnelse), men jeg kan ta en titt å se om jeg kan forbedre noe hvertfall :)